### DISCUSSION TITLE
Draw Path From Multiple Lines

### DISCUSSION THREAD

**Discussion by nigeljw3** [alandefreitas/matplotplusplus#290] [Q&A]
I am trying to create a plot with multiple lines. I specifically want to draw a path, but this can be simplified to drawing multiple lines. I have tried two different ways from the documentation:
https://alandefreitas.github.io/matplotplusplus/plot-types/line-plots/line-plot/

I am just trying to draw all lines in the path. If I use a set, it only lets me specify one set for X, and I can't seem to add a second set for the Y positions. Can someone clarify the correct way to do this?

std::set<std::vector<double>> linePointsX;
std::set<std::vector<double>> linePointsY;
...
matplot::plot(linePointsX, linePointsY);

This gives me the following error:
Error	C2664	'matplot::line_handle matplot::axes_type::plot(const std::vector<double,std::allocator<double>> &,const std::vector<double,std::allocator<double>> &,std::string_view)': cannot convert argument 1 from 'std::set<std::vector<double,std::allocator<double>>,std::less<std::vector<double,std::allocator<double>>>,std::allocator<matplot::vector_1d>>' to 'const std::vector<double,std::allocator<double>> &'

In other examples, it calls plot multiple times, and they overlap in the same plot, but when I try to call plot multiple times, it refreshes to only show the single line for each plot, even though I only call show once.

**Comment by nigeljw3** [comment]
I tried adding the lines in the same set

`std::vector<double> pointsX;
std::vector<double> pointsY;
pointsX.push_back(point1.x);
pointsX.push_back(point2.x);
pointsY.push_back(point1.y);
pointsY.push_back(point2.y);
linePoints.insert(pointsY);
linePoints.insert(pointsX);
matplot::plot(linePoints);`
        
But it doesn't work the same as adding them explicitly with plot(x1, y1, x2, y2, x3, y3, ...). This is the result:

![image](https://user-images.githubusercontent.com/504090/184135964-39075e4a-b093-461a-8456-2d46b849cc2c.png)

**Comment by alandefreitas** [comment]
The error is telling you the function expects `vector<double>`, not a `set<vector<double>>`.

**Reply by nigeljw3** [reply]
I understand the error, but do not understand what should be the expected function call to use a set for both X and Y. It only allows one like in this example from the documentation for line plots:

`#include <matplot/matplot.h>
#include <set>

int main() {
    using namespace matplot;

    std::set<std::vector<double>> Y = {
        {16, 5, 9, 4}, {2, 11, 7, 14}, {3, 10, 6, 15}, {13, 8, 12, 1}};
    plot(Y);

    show();
    return 0;
}`

**Reply by alandefreitas** [reply]
I'm not sure I understand the semantics of your sets, but I guess you want a shortcut for:

```cpp
std::set<std::vector> linePointsX;
std::set<std::vector> linePointsY;
// ...
auto it0 = linePointsX.begin();
auto end0 = linePointsX.end();
auto it1 = linePointsY.begin();
auto end1 = linePointsY.end();
for (; it0 != end0 && it1 && end1; ++it0, ++it1)
    matplot::plot(*it0, *it1);
```

?

I'm not sure `figure_type` has a function that does that for you. It never came up.

**Reply by nigeljw3** [reply]
Yeah, and the problem that I am running into is that if I call plot() multiple times, it displays each plot() call in sequence instead of overlapping them into one plot. Essentially the show() function is happening as a part of the call to plot() (even without calling show()) so it displays each one in sequence.

**Reply by nigeljw3** [reply]
I also can't save as an image because of this similar issue I mentioned here:
https://github.com/alandefreitas/matplotplusplus/issues/243

**Reply by alandefreitas** [reply]
> displays each plot() call in sequence

Then you need to call `hold(true)` (and read the docs)

**Comment by nigeljw3** [comment]
I want to be able to plot a set of lines with x and y values from sets/vectors.

**Comment by nigeljw3** [comment]
Like in the above sample using the set, how can I set explicit x values?
